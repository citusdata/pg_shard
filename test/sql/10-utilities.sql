-- ===================================================================
-- test utility statement functionality
-- ===================================================================

CREATE TABLE sharded_table ( name text, id bigint );
SELECT master_create_distributed_table('sharded_table', 'id');

-- cursors may not involve distributed tables
DECLARE all_sharded_rows CURSOR FOR SELECT * FROM sharded_table;

-- EXPLAIN support isn't implemented
EXPLAIN SELECT * FROM sharded_table;

-- PREPARE support isn't implemented
PREPARE sharded_query (bigint) AS SELECT * FROM sharded_table WHERE id = $1;
