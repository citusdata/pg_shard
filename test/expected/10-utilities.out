-- ===================================================================
-- test utility statement functionality
-- ===================================================================
CREATE TABLE sharded_table ( name text, id bigint );
SELECT master_create_distributed_table('sharded_table', 'id');
 master_create_distributed_table 
---------------------------------
 
(1 row)

-- cursors may not involve distributed tables
DECLARE all_sharded_rows CURSOR FOR SELECT * FROM sharded_table;
ERROR:  cannot perform distributed planning for the given query
DETAIL:  Utility commands are not supported in distributed queries.
-- EXPLAIN support isn't implemented
EXPLAIN SELECT * FROM sharded_table;
ERROR:  EXPLAIN commands on distributed tables are unsupported
-- PREPARE support isn't implemented
PREPARE sharded_query (bigint) AS SELECT * FROM sharded_table WHERE id = $1;
ERROR:  PREPARE commands on distributed tables are unsupported
